ngx-tools/src/utilities/object-deep-parse.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Helper function to parse an object with deep keys
+ *
+ * @param obj - the object
+ * @param path - the object path
+ * @return string value at the lowest layer
+ */
+
+export function objectDeepParse(object: { [key: string]: any } | string, keys: string | string[], defaultVal = object): any {
+  if (typeof object === 'string' || typeof object === 'boolean' || !object) {
```

Comment:
What are we trying to solve with this `if` statement?

Reply:
For input that's not an object with key value pair, no parsing needed, so it's returned.

Reply:
Why are we allowing something that isn't an object to be passed to a function called `deepObjectParse`?

Reply:
Technically `string` is a kind of `object`? 
If we only allow key value pair object passed in, all the references will have to check the type before passing in first. For this use case, it's because `changes.${controlName}.previousValue` or `changes.${controlName}.currentValue` could be string, boolean or object.